### ISSUE TITLE
InputField, Input, Label, Icon, Tooltip as compound components

### ISSUE THREAD

**Issue by hakimelek** [pricelinelabs/design-system#155]
I have been talking to @odmln and it seems that we had a confusing about building the `Input` component. Input should include, icon right, icon left, label, error states part of its behavior. Ideally it should be build as a [compound](https://github.com/ReactTraining/react-workshop/blob/master/subjects/CompoundComponents/lecture.js) component in my opinion. Meaning you can import `Input` and pass `Icon` (left or right) `Tooltip`, `Label` as children. 

Here is a demo of the Input Behavior (with outdated tooltip): 
![nov-17-2017 13-09-03](https://user-images.githubusercontent.com/1974993/32961927-833acf8e-cb98-11e7-8c60-779bcf4561e1.gif)

Input states:
![image](https://user-images.githubusercontent.com/1974993/32961837-257fd722-cb98-11e7-9e4c-e27b1e4f92e8.png)

Zeplin link: 
https://app.zeplin.io/project/596d11bc29e9db6303116850/screen/59ce7dfd4fb3d1a16d85589d

The tooltip should be able to show up up and down for the error states. #140 

before: 
```
<InputField label='' icon='' />
```

after: 
```
<InputField>
  <Icon />
  <Label />
  <Input />
  <Icon />
</InputField>
```

**Comment by jxnblk** [comment]
This is a great idea! React composition is super powerful, and this particular example would mean less custom API surface with the design system – if you know how React works, then you know how to move the icon from the left to the right side. I'd suggest we consider this with all of our composite components – Banner is another similar example: https://github.com/pricelinelabs/design-system/pull/111#discussion_r146889428

**Comment by byrekt** [comment]
Going to try to knock this out now, as using it for the type-ahead will be much cleaner that what I'm doing now.

**Comment by hakimelek** [comment]
Closing this since it should be addressed in #162.
